backend/src/controllers/post.controller.js

```
@@ -0,0 +1,164 @@
+export const createPost = asyncHandler(async (req, res) => {
+  const { userId } = getAuth(req);
+  const { content } = req.body;
+  const imageFile = req.file;
+
+  if (!content && !imageFile) {
+    return res
+      .status(400)
+      .json({ error: "Post must contain either text or image" });
+  }
+
+  if (post.user.toString() !== user._id.toString()) {
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)